presto-orc/src/test/java/com/facebook/presto/orc/TestReadBloomFilter.java

```diff
@@ -0,0 +1,133 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.orc;
+
+import com.facebook.presto.spi.predicate.NullableValue;
+import com.facebook.presto.spi.type.SqlVarbinary;
+import com.facebook.presto.spi.type.TimeZoneKey;
+import com.facebook.presto.spi.type.Type;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import io.airlift.units.DataSize;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static com.facebook.presto.memory.context.AggregatedMemoryContext.newSimpleAggregatedMemoryContext;
+import static com.facebook.presto.orc.OrcReader.MAX_BATCH_SIZE;
+import static com.facebook.presto.orc.OrcTester.Format.ORC_12;
+import static com.facebook.presto.orc.OrcTester.HIVE_STORAGE_TIME_ZONE;
+import static com.facebook.presto.orc.OrcTester.writeOrcColumnHive;
+import static com.facebook.presto.orc.TupleDomainOrcPredicate.ColumnReference;
+import static com.facebook.presto.orc.metadata.CompressionKind.SNAPPY;
+import static com.facebook.presto.spi.predicate.TupleDomain.fromFixedValues;
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.facebook.presto.spi.type.DoubleType.DOUBLE;
+import static com.facebook.presto.spi.type.IntegerType.INTEGER;
+import static com.facebook.presto.spi.type.SmallintType.SMALLINT;
+import static com.facebook.presto.spi.type.TinyintType.TINYINT;
+import static com.facebook.presto.spi.type.VarbinaryType.VARBINARY;
+import static com.facebook.presto.spi.type.VarcharType.VARCHAR;
+import static com.google.common.collect.Iterables.cycle;
+import static com.google.common.collect.Iterables.limit;
+import static com.google.common.collect.Lists.newArrayList;
+import static io.airlift.slice.Slices.utf8Slice;
+import static io.airlift.units.DataSize.Unit.MEGABYTE;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.testng.Assert.assertEquals;
+
+public class TestReadBloomFilter
+{
+    private static final TimeZoneKey TIME_ZONE = TimeZoneKey.getTimeZoneKey(HIVE_STORAGE_TIME_ZONE.getID());
+
+    @Test
+    public void test()
+            throws Exception
+    {
+        testType(TINYINT, ImmutableList.of(1L, 50L, 100L), 50L, 77L);
+        testType(SMALLINT, ImmutableList.of(1L, 5000L, 10_000L), 5000L, 7777L);
+        testType(INTEGER, ImmutableList.of(1L, 500_000L, 1_000_000L), 500_000L, 777_777L);
+        testType(BIGINT, ImmutableList.of(1L, 500_000L, 1_000_000L), 500_000L, 777_777L);
+
+        // Bloom Filters are not supported for DECIMAL, FLOAT, DATE, TIMESTAMP, and CHAR
```

Comment:
This comment is confusing to me. If bloom filters are not supported and this is test is specifically testing bloom-filter-enabled skipping, then how is the test passing for these types?

Reply:
As the comment says  Bloom Filters are not supported for DECIMAL, FLOAT, DATE, TIMESTAMP, and CHAR are not yet implemented in 
com.facebook.presto.orc.TupleDomainOrcPredicate.checkInBloomFilter
method for details as that does not supports these types yet, That supports TINYINT, SMALLINT, INTEGER, BIGINT, DOUBLE, VarcharType and VarbinaryType.

Reply:
@dilipkasana I must be missing something. It is confusing to me that although bloom filter is not supported for some types, the skipping still happens for these types. How does it work? I'd expect that for types that don't support bloom filters, stripe skipping won't work.

Reply:
In Presto Bloom Filter check support for Types TINYINT, SMALLINT, INTEGER, BIGINT, DOUBLE, VarcharType and VarbinaryType is supported only as per the method (om.facebook.presto.orc.TupleDomainOrcPredicate.checkInBloomFilter) , I have not investigated efforts to get the support done for  DECIMAL, FLOAT, DATE, TIMESTAMP, and CHAR  in that method.
So tests for TINYINT, SMALLINT, INTEGER, BIGINT, DOUBLE, VarcharType and VarbinaryType  are added and test like below will be failed :

testType(DATE, ImmutableList.of(new SqlDate(1), new SqlDate(5_000), new SqlDate(10_000)), 5_000L, 7_777L);
        testType(TIMESTAMP,
                ImmutableList.of(new SqlTimestamp(1, TIME_ZONE), new SqlTimestamp(500_000L, TIME_ZONE), new SqlTimestamp(1_000_000L, TIME_ZONE)),
                500_000L + HIVE_STORAGE_TIME_ZONE.getOffset(500_000L),
                777_777L + HIVE_STORAGE_TIME_ZONE.getOffset(777_777L));

        testType(REAL, ImmutableList.of(1.11f, 500_000.56f, 1_000_000.99f), (long) floatToIntBits(500_000.56f), (long) floatToIntBits(777_777.77f));
        testType(DOUBLE, ImmutableList.of(1.11, 500_000.55, 1_000_000.99), 500_000.55, 777_777.77);

Reply:
@dilipkasana Thanks for explaining. I understand it now. I got confused by seeing the comment in the middle of types. I didn't read carefully and assumed that types that appear after the comment are not supported. To avoid future confusion, I suggest to either move the comment to the end of the method or add `testTypeNotSupported` helper method that asserts that skipping never happens and use that to test the remaining types.